Realm/Realm/Logging/Logger.cs

```diff
@@ -144,6 +155,23 @@ protected override void LogImpl(LogLevel level, string message)
             }
         }
 
+        private class FileLogger : Logger
+        {
+            private readonly string _filePath;
+            private readonly Encoding _encoding;
+
+            public FileLogger(string filePath, Encoding encoding = null)
+            {
+                _filePath = filePath;
+                _encoding = encoding ?? Encoding.UTF8;
+            }
+
+            protected override void LogImpl(LogLevel level, string message)
+            {
+                System.IO.File.AppendAllText(_filePath, FormatLog(level, message) + Environment.NewLine, _encoding);
```

Comment:
This will be problematic because 1) it requires writing to the filesystem, which may be slow, meaning that the logger will negatively influence the performance of the system and 2) it's not taking any locks, so invoking the method from multiple threads concurrently will result in an IO exception being thrown.

Reply:
We discussed about this offline